src/core/lib/Mime.mjs

```diff
@@ -0,0 +1,310 @@
+        if (mimeObj.body && contType && contType.startsWith("multipart/")) {
+            if (!boundary) {
+                throw new OperationError("Invalid mulitpart section no boundary");
+            }
+            const sections = [];
+            for (const val of Mime._splitMultipart(mimeObj.body, boundary)) {
+                sections.push(Mime._parseMime(val));
+            }
+            if (sections.length) {
+                mimeObj.body = sections;
+            }
+        }
+        return mimeObj;
+    }
+
+    /**
+     * Executes a function on a mime object. These methods should modify the mimeObj.
+     *
+     * @param {Object} mimeObj
+     * @param {function} methods
+     * @param {boolean} recursive
+     */
+    static walkMime(mimeObj, method, recursive=true) {
+        const contType = Mime._extractField(mimeObj, "content-type");
+        method(mimeObj);
+        if (recursive && mimeObj.body && Array.isArray(mimeObj.body) && contType && contType.startsWith("multipart/")) {
+            mimeObj.body.forEach(function(obj) {
+                Mime.walkMime(obj, method);
+            });
+        }
+    }
+
+    /**
+     * Attempts to decode a mimeObj's data by applying appropriate character and content decoders based on the header data.
+     *
+     * @param {Object} mimeObj
+     */
+    static decodeMimeMessage(mimeObj) {
+        const contType = Mime._extractField(mimeObj, "content-type");
+        const contEnc = Mime._extractField(mimeObj, "content-transfer-encoding");
+        let charEnc = Mime._extractField(mimeObj, "content-type", "charset");
+        if (contType != null) {
+            if (!charEnc && contType.startsWith("text/")) {
+                charEnc = "us-ascii";
+            }
+        }
+        if (mimeObj.body && contEnc && typeof mimeObj.body === "string") {
+            mimeObj.body = Mime._decodeMimeData(mimeObj.body, charEnc, contEnc);
+        }
+    }
+
+    /**
+     * Takes a string and decodes quoted words inside them
+     * These take the form of:
+     * input "=?utf-8?Q?Hello_World!?="
+     * output "Hello World!"
+     *
+     * @param {string} input
+     * @param {string} type
+     * @returns {string}
+     */
+    static replaceEncodedWord(input) {
+        return input.replace(/=\?([^?]+)\?(Q|B)\?([^?]+)\?=/g, function (a, charEnc, contEnc, input) {
+            contEnc = (contEnc === "B") ? "base64" : "quoted-printable";
+            if (contEnc === "quoted-printable") {
+                input = input.replace(/_/g, " ");
+            }
+            return Utils.byteArrayToUtf8(Mime._decodeMimeData(input, charEnc, contEnc));
+        });
+    }
+
+    /**
+     * Breaks the header from the body and parses the header. The returns an
+     * object or null. The object contains the raw header, decoded body, and
+     * parsed header object.
+     *
+     * @param {string} input
+     * @returns {object}
+     */
+    static _splitParseHead(input) {
```

Comment:
The comment says this can return null but it doesn't seem to. Also, it might be easier to read if the object instantiations put each property on a new line.